File: backend/src/main/java/protasks/backend/Tag/Tag.java
package protasks.backend.Tag;

import com.fasterxml.jackson.annotation.JsonView;
import protasks.backend.Board.Board;
import protasks.backend.Task.Task;
import protasks.backend.TaskList.TaskList;
import protasks.backend.user.User;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Entity
public class Tag {    

    public interface TagBasicInfo{}
    public interface TagExtendedInfo{}

    @JsonView(Tag.TagBasicInfo.class)
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name="Id")
    private long id;

    @JsonView(Tag.TagBasicInfo.class)
    @Column(name="Name")
    private String name;

    @JsonView(Tag.TagBasicInfo.class)
    @Column(name="Color")
    private String color;

    @JsonView(Tag.TagExtendedInfo.class)
    @ManyToMany(mappedBy = "tag_ids")
    private List<Task> tasks ;

    @JsonView(Tag.TagExtendedInfo.class)
    @ManyToOne
    private Board board;

    @Column(name="write_date")
    @Temporal(TemporalType.TIMESTAMP)
    private Date write_date;

    @Column(name="create_date")
    @Temporal(TemporalType.TIMESTAMP)
    private Date create_date;

    public Tag() {
        this.create_date=new Date();
        this.write_date =new Date();
    }
    public Tag(String name, String color, Board board) {
        this.name=name;
        this.color=color;
        this.create_date=new Date();
        this.board=board;
        this.write_date =new Date();
    }
    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }


    public Board getBoard() {
        return board;
    }

    public void setBoard(Board board) {
        this.board = board;
    }

    public Date getWrite_date() {
        return write_date;
    }

    public void setWrite_date() {
        this.write_date = new Date();
    }

    public Date getCreate_date() {
        return create_date;
    }

    public List<Task> getTasks() {
        return tasks;
    }

    public void setTasks(List<Task> tasks) {
        this.tasks = tasks;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public void removeTask(Task t) {
        if (!this.tasks.isEmpty()){
            this.tasks.remove(t);
        }
    }
    

}

File: app/src/main/java/com/example/protasks/models/Tag.kt
package com.example.protasks.models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
class Tag {
    @Expose
    @SerializedName("id")
    private var id: Long = 0

    @Expose
    @SerializedName("name")
    private var name: String? = null

    @Expose
    @SerializedName("color")
    private var color: String? = null

    @Expose
    @SerializedName("tasks")
    private var tasks: List<Task?>? = null

    @Expose
    @SerializedName("board")
    private var board: Board? = null


    fun getId(): Long {
        return id
    }

    fun setId(id: Long) {
        this.id = id
    }

    fun getName(): String? {
        return name
    }

    fun setName(name: String?) {
        this.name = name
    }

    fun getTasks(): List<Task?>? {
        return tasks
    }

    fun setTasks(tasks: List<Task?>?) {
        this.tasks = tasks
    }

    fun getBoard(): Board? {
        return board
    }

    fun setBoard(board: Board?) {
        this.board = board
    }
    fun getColor(): String? {
        return color
    }

    fun setColor(color: String?) {
        this.color = color
    }

}

